File: Week2/BlueStaxks/2-1.cpp
#include <vector>
#include <iostream>
using namespace std;
vector<int> solution(vector<int> arr)
{
    vector<int> r;
    r.push_back(arr[0]);
    for (int i = 1; i < arr.size(); ++i)
        if (arr[i - 1] != arr[i]) //다를 경우만 넣기
            r.push_back(arr[i]);
    return r;
}

File: Week2/jihee/2-1.py
# 같은숫자는 싫어
def solution(arr):
    answer = []
    # stack을 이용해 바로 앞의 원소와 같다면 answer에 저장하지 않고,
    # 같지 않다면 answer에 저장
    for i in range(len(arr)):
        if answer: 
            tmp = answer.pop()
            # 바로 앞의 원소와 같지 않다면 answer에 저장
            if tmp != arr[i]:
                answer.append(tmp)
                answer.append(arr[i])
            # 바로 앞의 원소와 같다면 저장 x
            else:
                answer.append(tmp)
        else:
            answer.append(arr[i])

    return answer
    

if __name__ == "__main__":
    arr = [1,1,3,3,0,1,1]
    answer = solution(arr)
    print(answer)